pkg/devfile/adapters/common/interface.go

```diff
@@ -1,10 +1,15 @@
 package common
 
+import (
+	versionsCommon "github.com/openshift/odo/pkg/devfile/parser/data/common"
+)
+
 // ComponentAdapter defines the functions that platform-specific adapters must implement
 type ComponentAdapter interface {
 	Push(parameters PushParameters) error
 	DoesComponentExist(cmpName string) bool
 	Delete(labels map[string]string) error
+	Test(testcmd versionsCommon.DevfileCommand, show bool) error
```

Comment:
I think when we wrote the interface, there was a conscious effort to make it as low level as possible. So we should not use other structs like `versionsCommon.DevfileCommand`. It should pass the command name as `string` and pls check my other related comment on pkg/odo/cli/component/test.go

Reply:
Makes sense. Updated